fedimint-core/src/encoding/btc.rs

```diff
@@ -203,10 +203,13 @@ impl Decodable for bitcoin::Amount {
 impl Encodable for bitcoin::Address<NetworkUnchecked> {
     fn consensus_encode<W: std::io::Write>(&self, writer: &mut W) -> Result<usize, Error> {
         let mut len = 0;
-        len += NetworkLegacyEncodingWrapper(get_network_for_address(self.as_unchecked()))
-            .consensus_encode(writer)?;
+        len +=
+            NetworkLegacyEncodingWrapper(get_network_for_address(self)).consensus_encode(writer)?;
         len += self
             .clone()
+            // We need an `Address<NetworkChecked>` in order to get the script pubkey.
```

Comment:
`assume_checked` is only a risk if the address is untrusted (provided from the outsdie). If we're encoding something, it is "internal" data, and should have been validated already, so it's OK.

Reply:
Currently we implement `Encodable` and `Decodable` for unchecked addresses but not checked addresses, requiring us to re-check the network for addresses that, as you mention, are trusted to be already checked. What do you think about changing the implementations to be for checked addresses instead? That way, we could remove the few places where we're calling `into_unchecked()` or `as_unchecked()` when encoding data and likewise remove the need to re-check addresses when decoding them.

Reply:
I made #6550 to showcase this. However, since it requires the removal of `serde::Deserialize` for structs containing a checked address, it probably isn't worth it.

Reply:
@tvolk131 Maybe we should parametrize with aliases?


```
struct SomeAggregateType<V = NetworkUnchecked> {
  ...
  addresss: Address<V>,
  ...
}

type CheckedSomeAggregateType = SomeAggregateType<NetworkChecked>;

impl SomeAggregateType<NetworkUnchecked> {
  fn validate_network(...) -> ... {

  }
}
```

Database definitions can just use `CheckedSomeAggregateType` everywhere, while networking code the (unchecked) `SomeAggregateType`?

@tvolk131

Reply:
Feels like too much overhead to me tbh.